# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1513884966 18000
#      Thu Dec 21 14:36:06 2017 -0500
# Node ID b431e2719000a431df631af7f29f6fa14a245ad1
# Parent  4ca5c9f899d3934d2fc82c5191fd192c2b368139
* __opengl_info__.m: Make more than one attempt to get info from hidden figure.

diff --git a/scripts/plot/util/__opengl_info__.m b/scripts/plot/util/__opengl_info__.m
--- a/scripts/plot/util/__opengl_info__.m
+++ b/scripts/plot/util/__opengl_info__.m
@@ -102,25 +102,28 @@ function [info, msg] = gl_info ()
   for hf = figs.'
     info = fig_gl_info (hf);
     if (! isempty (info))
       break;
     endif
   endfor
 
   ## If no info yet, try open a figure to get the info.
-  if (isempty (info))
+  attempts = 1;
+  while (isempty (info) && attempts++ <= 3)
     ## Need to create a figure, place an OpenGL object, and force drawing.
     hf = figure ("position", [0,0,1,1], "toolbar", "none", "menubar", "none");
     hax = axes ();
-    pause (0.1);  # FIXME: Race condition means this delay may not always work.
+    ## FIXME: Race condition means this delay may not always work.
+    pause (0.1 * attempts);
     refresh (hf);
     info = fig_gl_info (hf);
     close (hf);
-  endif
+    attempts
+  endwhile
 
   if (isempty (info))
     msg = "__opengl_info__: can not obtain OpenGL information";
   endif
 
 endfunction
 
 
